### PR TITLE
Ripped single_key parameter from skat

### PR DESCRIPTION
**Pull request by jbloom22**
- updated parameter names
- fixed tests and docs

As with burden tests, explode should be used to handle variants with multiple keys

### CODE REVIEW

File: python/hail/tests/tests.py [hunk 1]
```diff
@@ -379,27 +379,25 @@ def test_dataset(self):
...
+            .annotate_variants_table(intervalsSkat, root="va.gene")
             .annotate_variants_table(weightsSkat, root="va.weight")
             .annotate_samples_table(phenotypesSkat, root="sa.pheno")
             .annotate_samples_table(covariatesSkat, root="sa.cov")
             .annotate_samples_expr("sa.pheno = if (sa.pheno == 1.0) false else " +
                                    "if (sa.pheno == 2.0) true else NA: Boolean"))
 
-        (skatVds.skat(variant_keys='va.genes',
-                      single_key=False,
+        (skatVds.skat(key_expr='va.gene',
```
Comment (cseed) [inline python/hail/tests/tests.py]: To keep the test the same, shouldn't you keep `product=True` in the annotate_variants_table above and explode here?

File: python/hail/tests/tests.py [hunk 2]
```diff
@@ -379,27 +379,25 @@ def test_dataset(self):
...
                       y='sa.pheno',
                       x='g.GT.nNonRefAlleles()',
                       covariates=['sa.cov.Cov1', 'sa.cov.Cov2'],
-                      weight_expr='va.weight',
                       logistic=False).count())
 
-        (skatVds.skat(variant_keys='va.genes',
-                      single_key=False,
+        (skatVds.skat(key_expr='va.gene',
+                      weight_expr='va.weight',
```
Comment (cseed) [inline python/hail/tests/tests.py]: Same.

File: src/test/scala/is/hail/methods/SkatSuite.scala [hunk 3]
```diff
@@ -163,7 +162,7 @@ class SkatSuite extends SparkSuite {
       .annotateSamples(TFloat64(), List("cov", "Cov1"), s => cov1Array(s.asInstanceOf[String].toInt))
       .annotateSamples(TFloat64(), List("cov", "Cov2"), s => cov2Array(s.asInstanceOf[String].toInt))
       .annotateSamples(TBoolean(), List("pheno"), s => phenoArray(s.asInstanceOf[String].toInt))
-      .annotateVariantsExpr("va.genes = [v.start % 2, v.start % 3].toSet") // three overlapping genes
+      .annotateVariantsExpr("va.gene = v.start % 3") // three genes
```
Comment (cseed) [inline src/test/scala/is/hail/methods/SkatSuite.scala]: Same as comment on other tests: shouldn't we test the product/explode/skat use case?

**Comment by jbloom22** [comment]
@cseed the tests compare skat and R which compute stats one group at a time, so I can compare product / explode / skat to product / explode / rskat, but that’s still only testing skat and product / explode independently.

I could get some numbers out from the previous version with single_key=True and hardcode those into a test of the new version.

I'd added testToKeyGsWeightRdd() specifically to check that the grouping by key checks out. I think it'd make more sense to expand that test to a case that uses product / explode.

**Comment by jbloom22** [comment]
See comment above. I've modified testToKeyGsWeightRdd to include interaction with explode. The test will fail until #2549 goes in since it used `toSet()`.

**Comment by jbloom22** [comment]
@catoverdrive I pushed more improvements just as you approved, would you mind taking a look at them as well (it's minimal). Sorry about that, thanks! I also want to clarify testing with Cotton before merging, let me know if you have any thoughts on the matter.

**Comment by jbloom22** [comment]
@cseed please see comments above and look at testToKeyGsWeightRdd(), let me know if you think it's enough. (I was able to rip out the SparseVector code that was supporting the GT case as well now that we've gone generic)

**Comment by jbloom22** [comment]
I'm merging this so Sali can use the new interface and better performance, and I am going to add an additional test in another PR.

**Pull request merged**